### PR TITLE
UG and DG Updates

### PR DESCRIPTION
**Pull request by Charles1026**
Add Hyperlinks to UG

Add `add`, `edit`, `delappt` and `consult` command descriptions to DG

Add keyboard shortcuts and Patient Details Panel descriptions to DG

Update Ui segment for DG

Add user stories, use cases and manual testing cases

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -226,6 +232,43 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
...
+`Medications` and `UpcomingAppointment`. Note that the `PastAppointment` field cannot be updated in this command,
+that is done in the [`Appt`](#pastappointment) and [`DelAppt`](#delappt-command ) commands.
+
+When `edit INDEX <<args>>` is inputted, the UI calls the `LogicManager` which then calls the `AddressBookParser` to parse the
+input. This then creates an instance of the `EditCommandParser` to parse the `INDEX` and `args` via the respective static
+`ParserUtil` functions. If duplicate parameters are inputted (e.g. `add n/Joe n/Mel`), only the last instance is taken,
+similar to how [`add`](#add-command), [`appt`](#pastappointment) and [`consult`](#consult-command)  are executed.
+
+The `EditCommandParser` will then create the corresponding `EditPersonDescriptor` object and then feed it to a
+`EditCommand` object it creates and returns. The `LogicManager` then executes the `EditCommand`, which adds creates a
```
Comment (Ryan-Tan00) [inline docs/DeveloperGuide.md:264]: typo here: `which adds creates`

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -428,20 +512,66 @@ Getting patients with an appointment on a specified date involves the following
...
-### New Add Command
-The new `Add` Command incorporates support for the necessary fields for a patient, namely they are the: `NextOfKin`,
-`PatientType`,`HospitalWing`, `FloorNumber`, `WardNumber` and `Medications` fields. The new command still follows the
-flow of the old command, as illustrated in the Activity Diagram below.
+### Patient Details Panel
+
+The Patient Details Panel provides a detailed view into the information of a specific patient. All the patient's personal 
+particulars and appointment details are reflected in this panel. The patient being viewed defaults to the first patient
+in the app, if present. Whenever the [`add`](#add-command) or [`edit`](#edit-command) is called on a patient, the patient 
+displayed switches to that patient in question. To manually change the person being viewed, the [`view`](#view-command)
```
Comment (Ryan-Tan00) [inline docs/DeveloperGuide.md:520]: typo here: `the view can be used`

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -476,25 +606,78 @@ The usage of the Add Command remains the same as before.
...
+      Use case resumes at step 1.
+
+
+**Use case: Edit a patient**
+
+**MSS**
+
+1. User requests to list persons
+2. CheckUp shows a list of persons
+3. User requests patient to edit
```
Comment (Ryan-Tan00) [inline docs/DeveloperGuide.md:660]: typo here: `requests patient to edit`

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -476,25 +606,78 @@ The usage of the Add Command remains the same as before.
...
+
+  Use case ends.
+
+* 3a. The information entered is invalid.
+    * 3a1. CheckUp shows an error message.
+
+      Use case resumes at step 3.
+
+* 3b. No information is entered to edit
+    * 3a1. CheckUp shows an error message.
```
Comment (Ryan-Tan00) [inline docs/DeveloperGuide.md:678]: typo here: `3a1`

File: docs/DeveloperGuide.md [hunk 5]
```diff
@@ -584,10 +767,38 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+
+* 2a. The list is empty.
+
+  Use case ends.
+
+* 3a. The given index is invalid.
+
+    * 3a1. CheckUp shows an error message.
+
+      Use case resumes at step 2.
```
Comment (Ryan-Tan00) [inline docs/DeveloperGuide.md:794]: typo here: `resumes at step 2`

File: docs/DeveloperGuide.md [hunk 6]
```diff
@@ -610,6 +821,35 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
```
Comment (Ryan-Tan00) [inline docs/DeveloperGuide.md:846]: typo here: `resumes at step 2`

File: docs/DeveloperGuide.md [hunk 7]
```diff
@@ -707,6 +979,56 @@ testers are expected to do more *exploratory* testing.
...
+
+    4. Other incorrect consult commands to try: `consult`, `consult x`, `...` (where x is larger than the list size)<br>
+       Expected: No past appointment is created. No upcoming appointment is removed. Error details shown in the status message.
+
+### View a patient
+
+1. View a patient while all persons are being shown
+
+    1. Prerequisites: List all persons using the `list` command. Multiple persons in the list. The second patient has
+       an upcoming appointment on the current day and the rest have no upcoming appointments.
```
Comment (Ryan-Tan00) [inline docs/DeveloperGuide.md:1020]: do not think it is required for `The second patient has an upcoming appointment on the current day and the rest have no upcoming appointments` to be added

**Review by Ryan-Tan00 — CHANGES_REQUESTED** [review]

**Review by kxrt — CHANGES_REQUESTED** [review]
The Ui diagram's missing `CountWindow`.

File: docs/DeveloperGuide.md (hunk 5 above)
Comment (Charles1026) [reply docs/DeveloperGuide.md:794]: for this i followed the given delete example, where use case resumes at step 2 if an invalid index is given

File: docs/DeveloperGuide.md (hunk 6 above)
Comment (Charles1026) [reply docs/DeveloperGuide.md:846]: similar to the above reply

File: docs/DeveloperGuide.md (hunk 7 above)
Comment (Charles1026) [reply docs/DeveloperGuide.md:1040]: This is for testing purposes to demonstrate that the upcoming appointment is reset using consult.

**Review by Ryan-Tan00 — APPROVED** [review]
LGTM

**Review by kxrt — APPROVED** [review]
Thanks for fixing it up!

**Pull request merged**